src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.PerAnalyzerState.cs

```diff
@@ -37,6 +37,38 @@ private class PerAnalyzerState
                 _currentlyAnalyzingDeclarationsMapPool = currentlyAnalyzingDeclarationsMapPool;
             }
 
+            public void Free()
+            {
+                lock (_gate)
+                {
+                    foreach (var analyzerStateData in _pendingEvents.Values)
+                    {
+                        FreeState_NoLock(analyzerStateData, _analyzerStateDataPool);
```

Comment:
does this actually clean up information inside of the state before putting it back to pool?